hw/ip/keymgr/doc/keymgr_testplan.hjson

```diff
@@ -0,0 +1,172 @@
+            - Check STATUS reg for each operation
+            - Check interrupts `op_done` is triggered when operation is done
+            - Check `err` and alert `operation_err` are triggered after invalid operation
+            - Check KMAC key, KMAC data and output SW data for correctness
+            '''
+      milestone: V1
+      tests: ["keymgr_smoke"]
+    }
+    {
+      name: cfgen_during_op
+      desc: '''
+            CFGEN is RO reg and it gates bunch of write access of other registers, which isn't
+            tested in common CSR tests.
+
+            Stimulus and checks:
+            Test command and reg access gated by CFGEN is ignored during operation.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: op_at_wipe_state
+      desc: '''
+            `StWipe` is a temporary state during init operation, which will set CSR CFGEN.
+
+            Stimulus and checks:
+            Test command and reg access gated by CFGEN is ignored during `StWipe`.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: hw_sideload
+      desc: '''
+            Keymgr contains HW sideload interfaces to output keys for KMAC, HMAC, AES.
+
+            Stimulus:
+            Generate a keymgr output to HW sideload interface, excising all the sideload interfaces.
+
+            Checks:
+            Verify the sideload data and status for correctness.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: init_n_start
+      desc: '''
+            Stimulus and checks:
+            Verify all combinations of `init` and `start`.
+            - If both are set in `StReset`, check `INVALID_OP` error is triggered.
+            - If it's in other state, `init` is ignored
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: direct_to_disabled_state
+      desc: '''
+            Stimulus and checks:
+            Directly go to `StDisabled` from any state and check `StDisabled` is entered correctly.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: lc_disable
+      desc: '''
+            Life cycle can disable keymgr and let keymgr wipe secret immediately
+
+            Stimulus:
+            Test life cycle disables keymgr in any state.
+
+            Checks:
+            - If keymgr is not initialized, check it can't be initialized until life cycle enables
+              keymgr.
+            - If keymgr is in a valid state after init, key output to KMAC is wiped immediately and
+              SW output will be invalid after OP is done
+            - If keymgr in disabled state, check the behavior is consistent with normal behavior.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: invalid_cmd
+      desc: '''
+            Key manager has some security checks for fault injection, and it triggers an
+            `INVALID_CMD` error. This isn't possible to be raised by SW.
```

Comment:
```suggestion
            Verify `keymgr` security countermeasures on fault injection.
```

The stimulus part explains why that this is done using forces. Checks part explains what countermeasures are expected.

Reply:
Thanks for the suggestion. Updated.